Habitica/src/main/java/com/habitrpg/android/habitica/ui/helpers/MarginDecoration.kt

```diff
@@ -10,6 +10,14 @@ class MarginDecoration(context: Context?) : RecyclerView.ItemDecoration() {
     private val margin: Int = context?.resources?.getDimensionPixelSize(R.dimen.grid_item_margin) ?: 0
 
     override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
-        outRect.set(margin, margin, margin, margin)
+        for (i in 0 until parent.childCount) {
+            val view: View = parent.getChildAt(i)
+            val position: Int = parent.getChildAdapterPosition(view)
+            val viewType: Int? = parent.adapter?.getItemViewType(position)
+            if (viewType == 2) {
+                outRect.set(margin, margin, margin, margin)
```

Comment:
The code itself looks good and I like the solution! Only thing I'm worried about is that now this decoration is not usable in other places anymore, because it will cause unexpected behavior. Can you add a new property on the `MarginDecoration` class like `zeroMarginViewTypes` that is a list of integers? And then change this code here to check that the `viewType` is not in that list before applying the margin. That way, when using this class you can specify that the `0` and `1` viewTypes should not get a margin and get the same behavior as you have right now. While still having the current behavior of it applying a margin for all items.
A potential optimization for this would then also be to check if `zeroMarginViewTypes` is null or empty and if yes, skip the whole viewType calculation and apply the margin directly.